test/tests/functional/pbs_snapshot_unittest.py

```diff
@@ -552,6 +557,90 @@ def test_snapshot_from_hook(self):
         self.assertTrue(rv)
         self.server.log_match(logmsg)
 
+    def snapshot_multi_mom_basic(self, obfuscate=False):
+        """
+        Test capturing data from a multi-mom system
+
+        :param obfuscate: take snapshot with --obfuscate?
+        :type obfuscate: bool
+        """
+        # Skip test if number of moms is not equal to two
+        if len(self.moms) != 2:
+            self.skipTest("test requires atleast two moms as input, " +
+                          "use -p moms=<mom 1>:<mom 2>")
+
+        mom1 = self.moms.values()[0]
+        mom2 = self.moms.values()[1]
+
+        host1 = mom1.shortname
+        host2 = mom2.shortname
+
+        rc = self.server.manager(MGR_CMD_DELETE, NODE, None, "")
+        self.assertEqual(rc, 0)
+
+        rc = self.server.manager(MGR_CMD_CREATE, NODE, id=host1)
+        self.assertEqual(rc, 0)
+
+        rc = self.server.manager(MGR_CMD_CREATE, NODE, id=host2)
+        self.assertEqual(rc, 0)
+
+        # Give the moms a chance to contact the server.
+        self.server.expect(NODE, {'state': 'free'}, id=host1)
+        self.server.expect(NODE, {'state': 'free'}, id=host2)
+
+        # Capture a snapshot with details from the remote moms
+        (_, snapdir) = self.take_snapshot(hosts=[host1, host2],
+                                          obfuscate=obfuscate)
+
+        # Check that snapshots for the 2 hosts were captured
+        host1_outtar = os.path.join(snapdir, host1 + "_snapshot.tgz")
+        host2_outtar = os.path.join(snapdir, host2 + "_snapshot.tgz")
+
+        self.assertTrue(os.path.isfile(host1_outtar))
+        self.assertTrue(os.path.isfile(host2_outtar))
+
+        # Unwrap the host snapshots
+        host1_snapdir = host1 + "_snapshot"
+        host2_snapdir = host2 + "_snapshot"
+        os.mkdir(host1_snapdir)
+        self.snapdirs.append(host1_snapdir)
+        os.mkdir(host2_snapdir)
+        self.snapdirs.append(host2_snapdir)
+        tar = tarfile.open(host1_outtar)
+        tar.extractall(path=host1_snapdir)
+        tar.close()
+        tar = tarfile.open(host2_outtar)
+        tar.extractall(path=host2_snapdir)
+        tar.close()
+
+        # Determine the name of the child snapshots
+        snap1_path = self.du.listdir(path=host1_snapdir, fullpath=True)
+        snap2_path = self.du.listdir(path=host2_snapdir, fullpath=True)
+        snap1_path = snap1_path[0]
+        snap2_path = snap2_path[0]
+
+        # Check that at least pbs.conf was captured on all of these hosts
+        self.assertTrue(os.path.isfile(os.path.join(snapdir, "pbs.conf")))
+        self.assertTrue(os.path.isfile(os.path.join(snap1_path,
+                                                    "pbs.conf")))
+        self.assertTrue(os.path.isfile(os.path.join(snap2_path,
+                                                    "pbs.conf")))
+
+        self.du.rm(host1_snapdir, recursive=True, force=True)
+        self.du.rm(host2_snapdir, recursive=True, force=True)
```

Comment:
Since you delete all directories in self.snapdirs in tearDown(), maybe just append these to self.snapdirs?

Reply:
I already am doing that earlier in the test, so these lines were not needed, I'll just remove them